### PR TITLE
add ConditionFailedError to {Set,}GoError()+test

### PR DESCRIPTION
**Pull request by tbg**
this adds some bits that were missing in #247

### CODE REVIEW

**Comment by tschottdorf** [comment]
@plaflamme may not have been an issue had we used 'oneof' for those union types as you suggested on our mailing list as I would assume we'd end up with setters and getters automagically. Maybe that's something you would want to look into?

File: storage/range_test.go [hunk 1]
```diff
@@ -1600,3 +1600,35 @@ func TestInternalMerge(t *testing.T) {
...
+			Bytes: value,
+		},
+		ExpValue: &proto.Value{
+			Bytes: []byte("moo"),
+		},
+	}
+	reply := &proto.ConditionalPutResponse{}
+	err := r.executeCmd(proto.ConditionalPut, args, reply)
+	if _, ok := err.(*proto.ConditionFailedError); err == nil || !ok {
+		t.Fatalf("expected ConditionFailedError, got %v with content %+v",
```
Comment (spencerkimball) [inline storage/range_test.go]: Use %T like:

    t.Fatalf("expected ConditionFailedError, got %T with content %+v", err, err)

Comment (spencerkimball) [inline storage/range_test.go]: Also might want to verify the actual value is "quack"

**Comment by spencerkimball** [comment]
LGTM

**Comment by plaflamme** [comment]
Ah, interesting. Sorry I missed that.

**Comment by tschottdorf** [comment]
It's one of the things that were bound to happen and now that @spencerkimball has jumped in, a thing of the past.

**Pull request merged**